lib/puppet_x/sqlserver/sql_connection.rb

```diff
@@ -90,25 +93,26 @@ def win32_exception
     end
 
     class ResultOutput
-      attr_reader :exitstatus, :error_message, :raw_error_message
 
-      def initialize(has_errors, error_message)
+      attr_reader :exitstatus, :error_message
+
+      def initialize(has_errors, error_message, connection)
         @exitstatus = has_errors ? 1 : 0
-        if error_message
-          @raw_error_message = error_message
-          @error_message = parse_for_error(error_message)
+
+        @error_message = (extract_messages(connection) || error_message)
+      end
+
+      def extract_messages(connection)
+        error_count = connection.nil? ? -1 : (connection.Errors.count - 1)
```

Comment:
How about an early return like this to avoid the nesting?

```ruby
return nil if connection.nil? || connection.Errors.count == 0

error_count = connection.Errors.count - 1
((0..error_count).map { |i| connection.Errors(i).Description.to_s }).join("\n")
```